ProfessorAssessor/View Controllers/HomeViewController.m

```diff
@@ -1,36 +1,180 @@
 
-- (IBAction)logout:(UIBarButtonItem *)sender;
+@property (nonatomic, strong) IBOutlet UITableView *filteredTableView;
+@property (nonatomic, strong) IBOutlet UITableView *sortedTableView;
+@property (nonatomic, strong) IBOutlet UISearchBar *searchBar;
+@property (nonatomic, strong) IBOutlet UILabel *schoolName;
+@property (nonatomic, strong) NSArray<Professor *> *professors;
+@property (nonatomic, strong) NSArray<Professor *> *filteredProfessors;
+@property (nonatomic, strong) NSArray<Professor *> *sortedProfessors;
 
 @end
 
 @implementation HomeViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+
+    [self fetchSchoolAndProfessors];
+
+    [self setUpRefreshControl];
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+
+    [self deselectTableViewRow];
 }
 
-- (IBAction)logout:(UIBarButtonItem *)sender {
-    [PFUser logOutInBackgroundWithBlock:^(NSError *_Nullable error) {
-        [[FBSDKLoginManager alloc] logOut];
+- (void)fetchSchoolAndProfessors {
+    __weak typeof(self) weakSelf = self;
 
-        [self displayLoginPage];
+    [Networker
+     fetchSchoolAndProfessorsWithCompletion:^(
+                                              PFObject *_Nullable object,
+                                              NSError *_Nonnull error) {
+        if (object) {
+            weakSelf.school = [School schoolFromPFObject:object];
+            weakSelf.schoolName.text = weakSelf.school.name;
+            
+            NSArray<Professor *> *professors = weakSelf.school.professors;
+            weakSelf.professors = [professors
+                                   sortedArrayUsingComparator:
+                                   ^NSComparisonResult(School *_Nonnull school1,
+                                                       School *_Nonnull school2) {
+                return [school1.name compare:school2.name];
+            }];
+            weakSelf.filteredProfessors = weakSelf.professors;
+            weakSelf.sortedProfessors = weakSelf.professors;
+
+            [weakSelf.filteredTableView reloadData];
+            [weakSelf.sortedTableView reloadData];
+            [weakSelf.sortedTableView.refreshControl endRefreshing];
+        }
     }];
 }
 
-- (void)displayLoginPage {
-    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
-    LogInOrSignUpViewController *viewController = [storyboard instantiateViewControllerWithIdentifier:@"LogInOrSignUpViewController"];
+- (void)setUpRefreshControl {
+    self.sortedTableView.refreshControl = [[UIRefreshControl alloc] init];
+
+    [self.sortedTableView.refreshControl addTarget:self action:@selector(fetchSchoolAndProfessors) forControlEvents:UIControlEventValueChanged];
+
+    [self.sortedTableView insertSubview:self.sortedTableView.refreshControl atIndex:0];
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
```

Comment:
Instead of sharing a delegate which is messy and error-prone, you could have a separate class that solely acts as the delegate. We can discuss this as an option.